### ISSUE TITLE
Base UI render prop: classes attributed to the trigger, not the rendered component

### ISSUE THREAD

**Issue by justinchen** [shadcn-ui/lint#20]
With Base UI (which shadcn/ui builds on), a trigger renders another component through a `render` prop. `@shadcn/lint` attributes the classes to the trigger rather than the rendered component, which produces both a wrong suggestion and a false positive.

### 1. Wrong component, unusable suggestion

```tsx
<DialogTrigger render={<Button />} className="bg-primary px-6" />
```

reports:

```
"bg-primary px-6" restyles <DialogTrigger> — use a variant (none defined)
```

The restyle is real, but the variants belong to `Button`, not `DialogTrigger`. Since the trigger defines none, the error tells the author there is no alternative, when `Button`'s variants are the answer.

### 2. False positive when the trigger renders a plain element

```tsx
<TooltipTrigger render={<span />} className="bg-primary px-6" />
```

is reported as a component restyle. Nothing in the design system was overridden — the classes land on a `span`.

### Expected

- Classes on a trigger with `render={<Component />}` are attributed to the rendered component, and suggestions come from that component's variants and sizes.
- A trigger rendering a plain element (`<span />`, `<div />`) is not a component restyle.

### Why it matters

This is the standard pattern for every trigger in a Base UI build (Dialog, Tooltip, Popover, DropdownMenu), so it is simultaneously the noisiest false positive and a systematically missed real catch.

Versions: `@shadcn/lint` 0.1.0, oxlint 1.83, Base UI.
